pkg/storage/pluginstore/grpc.go

```diff
@@ -0,0 +1,407 @@
+	opts := plugins.InsertOptions{
+		Collection: request.Collection,
+		Documents:  AsMapList(request.Documents),
+	}
+
+	err := m.impl.Insert(ctx, opts)
+	return &proto.InsertResponse{}, err
+}
+
+func (m *GServer) Patch(ctx context.Context, request *proto.PatchRequest) (*proto.PatchResponse, error) {
+	opts := plugins.PatchOptions{
+		Collection:     request.Collection,
+		QueryDocument:  AsMap(request.QueryDocument),
+		Transformation: AsOrderedMap(request.Transformation),
+	}
+
+	err := m.impl.Patch(ctx, opts)
+	return &proto.PatchResponse{}, err
+}
+
+func (m *GServer) Remove(ctx context.Context, request *proto.RemoveRequest) (*proto.RemoveResponse, error) {
+	opts := plugins.RemoveOptions{
+		Collection: request.Collection,
+		Filter:     AsMap(request.Filter),
+		All:        request.All,
+	}
+
+	err := m.impl.Remove(ctx, opts)
+	return &proto.RemoveResponse{}, err
+}
+
+func (m *GServer) Update(ctx context.Context, request *proto.UpdateRequest) (*proto.UpdateResponse, error) {
+	opts := plugins.UpdateOptions{
+		Collection: request.Collection,
+		Filter:     AsMap(request.Filter),
+		Upsert:     request.Upsert,
+		Document:   AsMap(request.Document),
+	}
+
+	err := m.impl.Update(ctx, opts)
+	return &proto.UpdateResponse{}, err
+}
+
+func NewPipeline(src []bson.D) []*proto.Stage {
+	pipeline := make([]*proto.Stage, len(src))
+	for i, srcStage := range src {
+		stage := &proto.Stage{
+			Steps: FromOrderedMap(srcStage),
+		}
+		pipeline[i] = stage
+	}
+
+	return pipeline
+}
+
+// ConvertBsonToPrimitives converts from bson primitives to pure Go primitives
+func ConvertBsonToPrimitives(src interface{}) interface{} {
+	switch t := src.(type) {
+	case bson.E:
+		rawValue := ConvertBsonToPrimitives(t.Value)
+		return map[string]interface{}{t.Key: rawValue}
+	case []bson.D:
+		raw := make([]interface{}, len(t))
+		for i, item := range t {
+			raw[i] = ConvertBsonToPrimitives(item)
+		}
+		return raw
+	case bson.D:
+		raw := make([]interface{}, len(t))
+		for i, item := range t {
+			raw[i] = ConvertBsonToPrimitives(item)
+		}
+		return raw
+	case []bson.M:
+		raw := make([]interface{}, len(t))
+		for i, item := range t {
+			raw[i] = ConvertBsonToPrimitives(item)
+		}
+		return raw
+	case bson.M:
+		raw := make(map[string]interface{}, len(t))
+		for k, v := range t {
+			raw[k] = ConvertBsonToPrimitives(v)
+		}
+		return raw
+	}
+
+	// Hope and pray we aren't sending something that protobuf doesn't like
+	// If we do, add another case above to handle that type
+	return src
+}
+
+// ConvertFloatToInt works around a weirdness in how numbers are represented
+// by structpb.Value, where integer values are stored in float64. When we
+// deserialize from protobuf, this walks the specified value, finds ints
+// that were encoded as floats, and converts them back to ints.
+func ConvertFloatToInt(src interface{}) interface{} {
+	switch tv := src.(type) {
+	case float64:
+		intVal := int(tv)
```

Comment:
I wonder if it's better to use int64 here just in case for some reason the float64 value doesn't fit into a int32